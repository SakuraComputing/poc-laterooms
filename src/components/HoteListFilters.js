import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByStarRating } from '../actions/filters';

export class HotelListFilters extends Component {

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

 onSortChange = (e) => {
    this.props.sortByStar(e.target.value);
 }

  render() {
    return (
      <div>
          <input
            type="text"
            placeholder="Hotel Name"
            value={this.props.filters.text}
            onChange={this.onTextChange}
            className="input-box"
          />
          <select className="select"
                  value={this.props.filters.sortBy}
                  onChange={this.onSortChange}
          >
            <option value="">Star Rating</option>
            <option value="Asc">Asc</option>
            <option value="Desc">Desc</option>
          </select>
      </div>
    )
  }
};

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByStar: (sortBy) => dispatch(sortByStarRating(sortBy)),
    // sortByStarDesc: () => dispatch(sortByStarRatingDesc()),
});

export default connect(mapStateToProps, mapDispatchToProps)(HotelListFilters);
