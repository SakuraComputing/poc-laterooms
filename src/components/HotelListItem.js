import React from 'react';

const HotelListItem = ({ name, starRating, facilities}) => {
    return (
        <div className="hotel-box">
            <h3>Hotel Name: {name}</h3>
            <div>Star Rating: {starRating}</div>
            <div>Facilities Include: {facilities.toString()}</div>
        </div>
    );
}

export default HotelListItem;
