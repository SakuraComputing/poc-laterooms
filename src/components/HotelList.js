import React from 'react';
import { connect } from 'react-redux';
import HotelListItem from './HotelListItem';

export const HotelList = (props) => {
    return (
        <div>
            {props.hotels.map((hotel) => {
                return <HotelListItem key={hotel.name} {...hotel} />;
            })}
        </div>
    );
}
const mapStateToProps = (state) => {
    return {
        hotels: state.hotels
    };
}

export default connect(mapStateToProps)(HotelList);
