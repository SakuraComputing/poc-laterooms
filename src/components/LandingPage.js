import React from 'react';
import HotelList from './HotelList';
import HotelListFilters from './HoteListFilters';
import '../App.css';

const LandingPage = () => {
    return (
        <div className="container">
            <h1>Late Rooms</h1>
            <div className="filter-bar">
                <HotelListFilters />
            </div>
            <HotelList />
        </div>
    );
}
export default LandingPage;
