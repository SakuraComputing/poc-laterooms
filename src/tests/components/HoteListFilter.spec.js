import React from 'react';
import { shallow } from 'enzyme';
import { HotelListFilters } from '../../components/HoteListFilters';

let setTextFilter, wrapper;


beforeEach(() => {
    setTextFilter = jest.fn();
    const filters = {
        name: '',
    }
    wrapper = shallow(
        <HotelListFilters
            filters = {filters}
            setTextFilter={setTextFilter}
        />
    )
});

describe('Hotel List Filter Compnonet', () => {
    it('should render expense list filter form correctly', () => {
        expect(wrapper).toMatchSnapshot()
    });
    // it('should render expense list filter with alt data correctly correctly', () => {
    //     wrapper.setProps({
    //        filters: altFilters
    //     });
    //     expect(wrapper).toMatchSnapshot()
    // });
    // it('should handle text change',() => {
    //     const value = 'New text';
    //     wrapper.find('input').at(0).simulate('change', {
    //         target: { value }
    //     });
    //     expect(setTextFilter).toHaveBeenLastCalledWith(value);
    // });
})




