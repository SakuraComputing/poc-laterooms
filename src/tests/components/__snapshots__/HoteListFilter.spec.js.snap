// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotel List Filter Compnonet should render expense list filter form correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HotelListFilters
    filters={
      Object {
        "name": "",
      }
    }
    setTextFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="input-box"
          onChange={[Function]}
          placeholder="Hotel Name"
          type="text"
        />,
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Any
          </option>
          <option
            value="Asc"
          >
            Asc
          </option>
          <option
            value="Desc"
          >
            Desc
          </option>
        </select>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-box",
          "onChange": [Function],
          "placeholder": "Hotel Name",
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value=""
            >
              Any
            </option>,
            <option
              value="Asc"
            >
              Asc
            </option>,
            <option
              value="Desc"
            >
              Desc
            </option>,
          ],
          "className": "select",
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Any",
              "value": "",
            },
            "ref": null,
            "rendered": "Any",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Asc",
              "value": "Asc",
            },
            "ref": null,
            "rendered": "Asc",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Desc",
              "value": "Desc",
            },
            "ref": null,
            "rendered": "Desc",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="input-box"
            onChange={[Function]}
            placeholder="Hotel Name"
            type="text"
          />,
          <select
            className="select"
            onChange={[Function]}
          >
            <option
              value=""
            >
              Any
            </option>
            <option
              value="Asc"
            >
              Asc
            </option>
            <option
              value="Desc"
            >
              Desc
            </option>
          </select>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-box",
            "onChange": [Function],
            "placeholder": "Hotel Name",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value=""
              >
                Any
              </option>,
              <option
                value="Asc"
              >
                Asc
              </option>,
              <option
                value="Desc"
              >
                Desc
              </option>,
            ],
            "className": "select",
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Any",
                "value": "",
              },
              "ref": null,
              "rendered": "Any",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Asc",
                "value": "Asc",
              },
              "ref": null,
              "rendered": "Asc",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Desc",
                "value": "Desc",
              },
              "ref": null,
              "rendered": "Desc",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
